std/algorithm/sorting.d

```diff
@@ -3748,58 +3748,46 @@ void topNIndex(alias less = "a < b", SwapStrategy ss = SwapStrategy.unstable,
               (Range r, RangeIndex index, SortOutput sorted = No.sortOutput)
 if (isRandomAccessRange!Range &&
     isRandomAccessRange!RangeIndex &&
-    hasAssignableElements!RangeIndex &&
-    isIntegral!(ElementType!(RangeIndex)))
+    hasAssignableElements!RangeIndex)
```

Comment:
As suggested I moved the specific ElementType checks and trigger a static assert error (and thus nice error message!) if an invalid type is passed by accident.